extensions/shared/register-jetpack-block.js

```diff
@@ -44,9 +57,23 @@ export default function registerJetpackBlock( name, settings, childBlocks = [] )
 		return false;
 	}
 
+	let blockTitle = settings.title;
+	const blockTags = [];
+
+	if ( requiredPlan ) {
+		blockTags.push( availableBlockTags.paid );
+	}
+	if ( betaExtensions.includes( name ) ) {
+		blockTags.push( availableBlockTags.beta );
+	}
+
+	if ( blockTags.length ) {
+		blockTitle = `${ blockTitle } (${ blockTags.join( ', ' ) })`;
```

Comment:
@akirk I'd be curious about your opinion about this. Do you think it would be useful to make a string like this translatable? I'm not sure something like "Simple Payments block (paid, beta)" is something that's always going to follow that format in every language. Maybe we should use placeholders in a translatable string here?

Reply:
Same code, but it now lives here: https://github.com/Automattic/jetpack/pull/14739/files#diff-bb769916a770ad52c514ba7ab6609e48R66

Just pinging here since the code here isn't really "Outdated", as github suggests.

Reply:
Hard to tell. From a translator perspective a string `%1$s (%2$s)` is always more confusing than helpful, and I'd only go that route when we get complaints that something looks bad. That said, I wonder if we don't have other options from a UI perspective to add "real" badges there vs doing title modification.

Reply:
> I wonder if we don't have other options from a UI perspective to add "real" badges there vs doing title modification.

That was our preferred route, but, sparing you the details, it wasn't going to be allowed to use badges 😞 This `(tags, list)` implementation was what we need to go with for now.